src/session.ts

```diff
@@ -195,13 +208,15 @@ function getMiddlewareAuthPathRegex(pathGlob: string) {
 async function redirectToSignIn() {
   const url = headers().get('x-url');
   const returnPathname = url ? getReturnPathname(url) : undefined;
+
   redirect(await getAuthorizationUrl({ returnPathname }));
 }
 
-async function getUser(options?: { ensureSignedIn: false }): Promise<UserInfo | NoUserInfo>;
-async function getUser(options: { ensureSignedIn: true }): Promise<UserInfo>;
-async function getUser({ ensureSignedIn = false } = {}) {
-  const session = await getSessionFromHeader('getUser');
+async function withAuth(options?: { ensureSignedIn: false }): Promise<UserInfo | NoUserInfo>;
+async function withAuth(options: { ensureSignedIn: true }): Promise<UserInfo>;
+async function withAuth({ ensureSignedIn = false } = {}) {
+  const session = await getSessionFromHeader('withAuth');
```

Comment:
why do we pass the name of the function to the `getSessionFromHeader` function?

Reply:
This is a bit of a legacy thing since we used to have more than one function that calls `getSessionFromHeader` and we wanted to be able to include the caller in the thrown error. I'll remove it since it's not used anymore.